lib/api/CDataSummarizationJsonSerializer.cc

```diff
@@ -396,35 +404,30 @@ CRetrainableModelJsonDeserializer::bestForestFromJsonStream(const core::CDataSea
         }
         forest->push_back(nodes);
     }
+
     return forest;
 }
 
-CRetrainableModelJsonDeserializer::TNodeVecVecUPtr
-CRetrainableModelJsonDeserializer::bestForestFromDocumentCompressed(
-    const core::CDataSearcher::TIStreamP& istream) {
-    rapidjson::IStreamWrapper isw{*istream};
-    rapidjson::Document doc;
-    doc.ParseStream(isw);
-    if (doc.HasMember(CInferenceModelDefinition::JSON_COMPRESSED_INFERENCE_MODEL_TAG) == false ||
-        doc[CInferenceModelDefinition::JSON_COMPRESSED_INFERENCE_MODEL_TAG].IsObject() == false) {
-        LOG_ERROR(<< "Field " << CInferenceModelDefinition::JSON_COMPRESSED_INFERENCE_MODEL_TAG
-                  << " not found or is not an object.");
+CRetrainableModelJsonReader::TNodeVecVecUPtr
+CRetrainableModelJsonReader::bestForestFromDocumentCompressed(TIStreamSPtr istream) {
+    if (istream == nullptr) {
         return nullptr;
     }
-    auto& compressedDataSummarization =
-        doc[CInferenceModelDefinition::JSON_COMPRESSED_INFERENCE_MODEL_TAG];
-    if (compressedDataSummarization.HasMember(CInferenceModelDefinition::JSON_DEFINITION_TAG) == false ||
-        compressedDataSummarization[CInferenceModelDefinition::JSON_DEFINITION_TAG]
-                .IsString() == false) {
-        LOG_ERROR(<< "Field " << CInferenceModelDefinition::JSON_DEFINITION_TAG
-                  << " not found or is not a string.");
-        return nullptr;
+    try {
+        rapidjson::IStreamWrapper isw{*istream};
+        rapidjson::Document doc;
+        doc.ParseStream(isw);
+        auto compressedDataSummarization = ifExists(
+            CInferenceModelDefinition::JSON_COMPRESSED_INFERENCE_MODEL_TAG, getObject, doc);
+        TStreamBuffer buffer{ifExists(CInferenceModelDefinition::JSON_DEFINITION_TAG,
+                                      getString, compressedDataSummarization),
+                             ifExists(CInferenceModelDefinition::JSON_DEFINITION_TAG,
+                                      getStringLength, compressedDataSummarization)};
+        return bestForestFromJsonStream(decompressStream(buffer));
+    } catch (...) {
```

Comment:
Same as previous comment regarding eating low level errors.

Reply:
Done in [f2c3588](https://github.com/elastic/ml-cpp/pull/1894/commits/f2c35881e9ec52f95d64cd4ba16bc45c3712fed3).